### DISCUSSION TITLE
reconcile

### DISCUSSION THREAD

**Discussion by lishine** [solidjs/solid#1264] [Q&A]
I can't understand what it is at all, from the documentation and not even from some examples I found.
Would be greatfull for explanation.

In general, though it took me a short time to read  the documentation, it seems to be lacking a lot about:
1. understanding each api
2. understanding how to use signal/props/createStore reactivity.
specifically how to make a primitive to be responsive to a data.
dealing now with the createVirtualizer from tanstack.
With React it is simple - after new render, the hook will receive the new data... 

Coming from React ecosystem (experienced with mobx and such)

**Comment by atk** [comment]
The documentation is momentarily being rewritten and feedback like yours is really valuable.

`reconcile` is meant for cases where you basically get a new object with some changes for your store and need to only apply the changes. One example would be a JSON answer from your server that changes gradually over time. Putting the response in a store makes its properties granular responsive, but in order to process the updates that contain the same root object, you need reconcile.

Basically, the main difference between react and solid is that in react, running code as a reaction is the default and to run something once, you need to opt out of its reactivity, like a car with the pedal always on the metal and you need to be on the brake all the time. On the other hand, in solid, reactivity is opt-in on all levels, akin to a normal car where you are fully in control. Code runs only once if called and you can register reactive effects and memos. These require all getters to be called as function (directly, like with signals, or via proxy, like in stores). If you keep that in mind, most of the rest should be pretty straightforward.

**Reply by lishine** [reply]
Thank you!
So reconcile is like deep merge? I am still not on it :) (maybe need a specific useful example).

Yes, I understand the opt-in/opt difference, thanks for phrasing that. I like the idea, but will have to actually use it in order to evaluate. After I have used mobx,  later on still preferred other non proxy redux/atom/immutable state managements, maybe because of some specifics React related.
I understand the idea, but not sure if the current api is good, so evaluating...

Currently the disadvantage that I see , with third party primitive, like the createVirtual, it seems like I have to look into it's source code in order to understand how it treats it's params. What params I have to feed it in order for it to be reactive.

Eventually, ok I don't think about implicit stuff like closures or multiple function calling here, but I have to consider the reactivity of a variable - to connect the reactivity chain in mind. Especially when I read code.
We implement reactivity within js limits

**Comment by lishine** [comment]
I got an answer for the createVirtualizer hook, they suggested a getter.
That is something you can also add emphasis in doc. I saw mentions of it in the doc but didn't actually connect the dots of why would I use it, as solution for what problem... 
React non mobx people usually don't use it so they don't know what it is for.
https://github.com/TanStack/virtual/pull/304#issuecomment-1267050145
